### ISSUE TITLE
Week 2 (ML focus)

### ISSUE THREAD

**Issue by adnan-sami-dev** [SMHub/job_prep#2]
> Preprocessing:

Handling NA values, outlier treatment, data normalization 
One hot encoding, label encoding 
Feature engineering 
Train test split 
Cross validation 

> Model Building:

o Types of ML: Supervised, Unsupervised 
o Supervised: Regression vs Classification 
o Linear models 
- Linear regression, logistic regression 
- Gradient descent 
o Nonlinear models (tree-based models) 
- Decision tree 
- Random forest 
- XGBoost 
o Model evaluation 
- Regression: Mean Squared Error, Mean Absolute Error, MAPE 
- Classification: Accuracy, Precision-Recall, F1 Score, ROC Curve, 
Confusion matrix 
- Hyperparameter tunning: GridSearchCV, RandomSearchCV  
- Unsupervised: K means, Hierarchical clustering, Dimensionality reduction (PCA) 

> ML Projects with Deployment
